src/api/goods.ts

```diff
@@ -5,15 +5,32 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
 
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL)
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        return Promise.reject(
+          new Error(response.statusText),
+        );
+      }
+
+      if (!response.headers.get('content-type')?.includes('application/json')) {
+        return Promise.reject(
+          new Error('Content type is not supported'),
+        );
+      }
+
+      return response.json();
+    });
 }
 
 export const get5First = () => {
   return getAll()
-    .then(goods => goods); // sort and get the first 5
+    .then(goods => goods
+      .sort((firstGood, nextGood) => firstGood.name
+        .localeCompare(nextGood.name))
+      .slice(0, 5)); // sort and get the first 5
```

Comment:
```suggestion
      .slice(0, 5));
```

Reply:
now such comments can be removed, your code should be understandable without explanation comments